x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/settings/update_button.tsx

```diff
@@ -156,40 +157,68 @@ export const UpdateButton: React.FunctionComponent<UpdateButtonProps> = ({
     setIsUpdateModalVisible(false);
     setIsUpgradingPackagePolicies(true);
 
-    await installPackage({ name, version, title });
-
-    upgradePackagePoliciesMutation.mutate(
-      {
-        // Only upgrade policies that don't have conflicts
-        packagePolicyIds: packagePolicyIds.filter(
-          (id) => !dryRunData?.find((dryRunRecord) => dryRunRecord.diff?.[0].id === id)?.hasErrors
-        ),
-      },
-      {
-        onSuccess: () => {
-          notifications.toasts.addSuccess({
-            title: toMountPoint(
-              <FormattedMessage
-                id="xpack.fleet.integrations.packageUpdateSuccessTitle"
-                defaultMessage="Updated {title} and upgraded policies"
-                values={{ title }}
-              />,
-              { theme$ }
-            ),
-            text: toMountPoint(
-              <FormattedMessage
-                id="xpack.fleet.integrations.packageUpdateSuccessDescription"
-                defaultMessage="Successfully updated {title} and upgraded policies"
-                values={{ title }}
-              />,
-              { theme$ }
-            ),
-          });
+    const hasUpgraded = await installPackage({ name, version, title, isUpgrade: true });
+    //  If install package failed do not upgrade package policies
+    if (!hasUpgraded) {
+      setIsUpgradingPackagePolicies(false);
+      return;
+    }
 
-          navigateToNewSettingsPage();
-        },
-      }
+    // Only upgrade policies that don't have conflicts
+    const packagePolicyIdsToUpdate = packagePolicyIds.filter(
+      (id) => !dryRunData?.find((dryRunRecord) => dryRunRecord.diff?.[0].id === id)?.hasErrors
     );
+
+    if (!packagePolicyIdsToUpdate.length) {
+      setIsUpgradingPackagePolicies(false);
+      navigateToNewSettingsPage();
+    }
+
+    try {
+      await upgradePackagePoliciesMutation.mutateAsync({
```

Comment:
(nit) If preferred, we can use the callbacks provided by `mutate` instead of `mutatAsync` and avoid the try/catch block. Very minor difference though imho - especially when there's not much logic inside the try/catch.

Reply:
Sure I can make the change, how this work when we have multiple mutation dependent on each other? it not going to be a callback hell?